app/services/otp_rate_limiter.rb

```diff
@@ -41,7 +42,7 @@ def otp_last_sent_at
   end
 
   def rate_limiter
-    @rate_limiter ||= RateLimiter.new(rate_limit_type: :phone_otp, target: rate_limit_key)
+    @rate_limiter ||= RateLimiter.new(rate_limit_type: @limit_type, target: rate_limit_key)
```

Comment:
Minor: In the `attr_reader` below, we're aliasing other instance variables `phone`, `user`, and `phone_confirmed` so we can call them as methods. We could add `limit_type` to that for consistency, and call this as:

```suggestion
    @rate_limiter ||= RateLimiter.new(rate_limit_type: limit_type, target: rate_limit_key)
```